akka-stream/src/main/scala/akka/stream/ThrottleMode.scala

```diff
@@ -24,12 +24,12 @@ object ThrottleMode {
   /**
    * Java API: Tells throttle to make pauses before emitting messages to meet throttle rate
    */
-  def shaping = Shaping
+  def shaping: ThrottleMode = Shaping
 
   /**
    * Java API: Makes throttle fail with exception when upstream is faster than throttle rate
    */
-  def enforcing = Enforcing
+  def enforcing: ThrottleMode = Enforcing
```

Comment:
Needs to return `Shaping` and `Enforcing` respectively. 

That's not that bad to type them with `Shaping` and `Enforcing`. Probably even desirable. 

These methods are there to help the Java API to get an instance of this two Scala objects. So, it's kind of correct if we explicitly return the real types. 

An equivalent code in Scala would be using the two object directly and therefore typing on the concrete types.